### PR TITLE
fix state-dependent tests

### PR DESCRIPTION
**Pull request by alanag13**
The nightly build was failing due to some tests that only worked when run in a certain order. Going through all tests and running them individually exposed which tests needed adjusting.

### CODE REVIEW

**Review by timabrmsn — APPROVED** [review]

File: tests/cmds/test_cases.py [hunk 1]
```diff
@@ -209,8 +202,12 @@ def test_show_with_include_file_events_calls_file_events_get_all_with_expected_p
     cli_state.sdk.cases.file_events.get_all.assert_called_once_with(1)
 
 
-def test_show_when_py42_raises_exception_prints_error_message(runner, cli_state, error):
-    cli_state.sdk.cases.file_events.get_all.side_effect = Py42NotFoundError(error)
+def test_show_when_py42_raises_exception_prints_error_message(
```
Comment (unparalleled-js) [inline tests/cmds/test_cases.py:205]: For `CASE_DETAILS`, can we make that the actual response for getting a case? And maybe rename it to `CASE_DETAILS_RESPONSE_TEXT` or something similar

**Comment by unparalleled-js** [comment]
can we make the text for the `custom_error` fixture JSON text?

File: tests/cmds/test_cases.py [hunk 2]
```diff
@@ -273,9 +270,9 @@ def test_file_events_add_calls_add_event_with_expected_params(runner, cli_state)
```
Comment (unparalleled-js) [inline tests/cmds/test_cases.py:270]: On line 229, it seems to me like the wrong method is being mocked ... I don't understand what is going. `get_all` does not get called in `show`, as far as I can tell.

shouldnt we be mocking `cases.get()` instead?

**Comment by unparalleled-js** [comment]
Consider it approved on my end, but I found some other oddities and ways that would make the tests less likely to break from py42 changes.

File: tests/cmds/test_cases.py (hunk 2 above)
Comment (kiran-chaudhary) [reply tests/cmds/test_cases.py:270]: the test method at line 229, `test_show_prints_expected_data_with_include_file_events_option` ,  refers to `file-events` `get_all`, so its fine. When `--include-file-events` is passed we want all the events listed against the case.

**Review by kiran-chaudhary — APPROVED** [review]
Changes LGTM. 

Honestly, Looking at the refactor, I am not able to tell what issue has been resolved, some more detail in the description, a link to the nightly build execution, would have helped to understand the scenario.

Having said that, I will try to find the nightly build execution, that shouldn't be hard assuming it should be in last few executions somewhere.

**Comment by kiran-chaudhary** [comment]
> Changes LGTM.
> 
> Honestly, Looking at the refactor, I am not able to tell what issue has been resolved, some more detail in the description, a link to the nightly build execution, would have helped to understand the scenario.
> 
> Having said that, I will try to find the nightly build execution, that shouldn't be hard assuming it should be in last few executions somewhere.

Please ignore this. I started my day with reviews and show the slack conversation later,  it helps to understand the investigation part.

**Pull request merged**